docs/proposals/distributed-tracing.md

```diff
@@ -0,0 +1,315 @@
+
+  * **Instrumentation Focus**: This component serves as the entry point for inference requests, providing
+optimized routing and load balancing.
+
+  * **Requests to Instrument**:
+    - Request entry through the gateway
+    - EPP routing decision execution
+    - Request forwarding to selected model instance
+    - Request duration and basic success/failure status
+
+  * **Key Spans and Attributes**:
+    - Gateway request receipt and processing
+    - EPP routing decision and endpoint selection
+    - Request forwarding to selected model instance
+    - Request completion status
+
+  * **Benefit**: Provides complete end-to-end visibility from the Kubernetes ingress layer through 
+the intelligent routing decisions, enabling validation of advanced scheduling optimizations like
+KV-cache aware routing and identification of gateway-level performance bottlenecks.
+
+### Unified and Opt-in Feature
+
+Given that llm-d is designed as a solution of several individual components deployable via a single Helm chart, the tracing feature will be
+implemented as an opt-in feature configured through the llm-d-deployer.
+
+This approach provides:
+
+* **Simplicity**: Users enable distributed tracing across their entire llm-d stack with minimal configuration.
+
+* **Operational Consistency**: Centralized configuration ensures all components send traces to a designated collector, simplifying trace correlation
+and analysis.
+
+* **Granular Control**: While top-level enablement is unified, individual components retain fine-grained controls.
+
+* **Scalability**: Single collector endpoint configuration supports scalable trace ingestion and forwarding to various observability backends.
+
+### Trace Context Propagation
+
+Even when tracing is disabled for individual components, **trace context propagation must be maintained** to preserve end-to-end trace continuity. Components with disabled tracing should still:
+
+* Extract incoming trace context from HTTP/gRPC headers
+* Propagate trace context to downstream service calls
+* Include trace context in outgoing HTTP/gRPC headers
+
+This ensures that disabling tracing for a single component doesn't break the distributed trace chain. OpenTelemetry provides lightweight context propagation that operates independently of span creation and export.
+
+**Performance Impact:**
+Context propagation has minimal overhead when tracing is disabled:
+- OpenTelemetry uses no-op propagators by default when unconfigured
+- Propagators are stateless and designed to avoid runtime allocations
+- Header extraction/injection operations are constant-time
+- No spans are created or exported, only context headers are passed through
+
+### Configuration
+
+```yaml
+tracing:
+  enabled: false
+  otelCollectorEndpoint: "http://otel-collector:4317"
+  apiToken: ""
+  samplingRate: 0.1
+  detailedTracing: false
+  # Always propagate trace context, even when component tracing is disabled
+  alwaysPropagateContext: true
```

Comment:
```suggestion
```
I assume that would be the case if traceing is enabled, but no endpoint to export the data is set?